test/webserver.js

```diff
@@ -80,7 +80,11 @@ WebServer.prototype = {
   _handler: function (req, res) {
     var url = req.url.replace(/\/\//g, '/');
     var urlParts = /([^?]*)((?:\?(.*))?)/.exec(url);
-    var pathPart = decodeURI(urlParts[1]), queryPart = urlParts[3];
+    // guard against directory traversal attacks,
+    // e.g. /../../../../../../../etc/passwd
```

Comment:
This example doesn't seem to agree with the regular expression used below; did you intend this to be `../../../../../../../etc/passwd` instead?
Or is this comment actually correct, and the regular expression wrong?

Reply:
`/../../../../../../../etc/passwd` is the path component of the URI. It will always start with `/`.

In theory, `path.normalize()` will turn `/../../../../../../../etc/passwd` into `/etc/passwd` as the path, which should fix pdf.js's current problem (as far as I can tell). The way the code parses URLs using regular expressions instead of using built-in libraries is pretty error-prone, so this was just a failsafe. If you would prefer, I can simply make it `path.normalize(decodeURI(urlParts[1]))`.

Reply:
If the comment is correct, doesn't that mean that the regular expression is wrong though?
Assuming that you actually want to transform `/../../../../../../../etc/passwd` into `/etc/passwd`, then the regexp needs to read `/^([\/\\]\.\.)+/` to work as expected.

Reply:
Let's just go with the path.normalize. I lean towards avoiding regex when possible.